src/libpriv/rpmostree-postprocess.c

```diff
@@ -493,20 +501,21 @@ convert_var_to_tmpfiles_d_recurse (GOutputStream *tmpfiles_out,
             return glnx_throw (error, "Failed to find group '%u' for %s", stbuf.st_gid, dent->d_name);
           g_string_append_printf (tmpfiles_d_buf, " %s %s - -", user, group);
 
-          /* Push prefix */
-          g_string_append_c (prefix, '/');
-          g_string_append (prefix, dent->d_name);
+          if (filetype_c == 'd')
+            {
+              /* Push prefix */
+              g_string_append_c (prefix, '/');
+              g_string_append (prefix, dent->d_name);
 
-          if (!convert_var_to_tmpfiles_d_recurse (tmpfiles_out, dfd, pwdb, prefix,
-                                                  cancellable, error))
-            return FALSE;
+              if (!convert_var_to_tmpfiles_d_recurse (tmpfiles_out, dfd, pwdb, prefix,
+                                                      cancellable, error))
+                return FALSE;
 
-          /* Pop prefix */
-          {
-            char *r = memrchr (prefix->str, '/', prefix->len);
-            g_assert (r != NULL);
-            g_string_truncate (prefix, r - prefix->str);
-          }
+              /* Pop prefix */
+              const char *r = memrchr (prefix->str, '/', prefix->len);
+              g_assert (r != NULL);
+              g_string_truncate (prefix, r - prefix->str);
```

Comment:
Minor: not new, though we can do the same thing as in libostree for SELinux prefixing and just remember `prefix->len` and truncate back to that here, instead of calling `memrchr`.